### ISSUE TITLE
Temporal label drift affects 8 of 9 ADME assays, but the time-split tests structural novelty — not value extrapolation

### ISSUE THREAD

**Issue by srijitseal** [srijitseal/polaris#33]
# Temporal label drift affects 8 of 9 ADME assays, but the time-split tests structural novelty — not value extrapolation

## Summary

Two questions raised in review of the temporal split: (1) is there label drift over time, and does it vary by assay (Engkvist)? (2) is "extrapolating to unseen property values" different from "extrapolating to an unseen chemical series whose values you have seen before" (Rodríguez)?

Both hold here. Measured values drift over time in **8 of the 9 ADME assays** — only mouse plasma protein binding (MBPB) is stable — confirming the "some but not all assays" pattern. And the temporal split, while it does move test molecules into less familiar chemical space, almost never pushes them beyond the training value range (≤0.9% of test molecules across all assays). Genuine value extrapolation is produced only by the target-value split (91.5% of LogD test molecules beyond range). So temporal deployment here is a *distribution-shift-within-the-seen-range* problem, not the "no potent compounds in training" problem — those are distinct generalization tests, exercised by different splits.

## Method

**No new splits were created.** This analysis reuses the cross-validation fold assignments already saved by the project: the time-based split (notebook 2.04), and — for the comparison anchors only — the chemical-cluster split (2.03) and the target-value split (2.05).

**Temporal proxy.** Each molecule has an ordinal registration index that increases with the order compounds entered the dataset; we use this index as a stand-in for calendar time. The time-based split is an *expanding window*: molecules are sorted by this index and cut into five consecutive chunks. The earliest chunk is training-only; each later chunk becomes, in turn, a test window whose training set is everything registered earlier. These are exactly the windows defined in notebook 2.04 — we read them, we do not recompute them.

**Value transform.** Measured values are log-transformed exactly as in modeling — base-10 logarithm of (value + 1) after flooring negligible values — except LogD, which is left as-is because it is already a logarithmic quantity.

### Analysis 1 — Label drift over time, per assay

For each assay, every measured molecule is placed in its temporal window (the 2.04 windows above), and two complementary statistics are computed:

- **Trend strength** — the Spearman rank correlation between a molecule's ordinal index and its measured value, over all molecules in the assay. This runs from −1 to +1; positive means values tend to rise in later compounds, negative means they fall. We also report its p-value.
- **Distribution shift** — the two-sample Kolmogorov–Smirnov statistic comparing the value distribution of the earliest window to that of the latest window. It runs from 0 (identical distributions) to 1 (no overlap).

Drift is flagged as **appreciable** only when it is both statistically reliable (p < 0.05) *and* of non-trivial size (absolute correlation ≥ 0.10). The size threshold matters because, with thousands of molecules per assay, almost any non-zero correlation is "statistically significant" on its own.

### Analysis 2 — Value extrapolation vs structural novelty

A test set can be "out of distribution" in two unrelated ways, and we quantify each separately:

- **Structural novelty** (unseen chemical series) — test molecules are chemically unlike anything in training, even if their measured values are ordinary. Measured as the median, over test molecules, of the distance from each test molecule to its single nearest training molecule, where distance is 1 − Tanimoto similarity on 2048-bit ECFP4 fingerprints (0 = identical fingerprint, 1 = no shared substructure bits).
- **Value extrapolation** (unseen property values) — test molecules carry measured values outside anything seen in training (e.g., more potent or lower-clearance than any training example), regardless of how familiar their structure is. Measured as the percentage of test molecules whose value lies above the training maximum or below the training minimum.

Both are computed for the time-split of every assay, and — to anchor the two axes — for all three split strategies on LogD.

### 1. Label drift is present in 8 of 9 assays (only MBPB stable)

Sorted by trend strength. Positive correlation = values rise in later compounds; "Δ median" is the change in (transformed) median from the earliest to the latest temporal window.

| Assay | n | Trend (Spearman) | p-value | Δ median (early→late) | Distribution shift (KS) | Appreciable drift |
|---|---|---|---|---|---|---|
| Caco-2 Permeability Efflux | 3,777 | +0.46 | 1e-199 | +0.36 | 0.46 | **yes** |
| Caco-2 Permeability Papp A>B | 3,773 | −0.39 | 5e-135 | −0.59 | 0.43 | **yes** |
| MLM CLint (mouse microsomal clearance) | 5,692 | −0.38 | 9e-191 | −0.50 | 0.35 | **yes** |
| KSOL (kinetic solubility) | 7,298 | +0.24 | 3e-94 | +0.55 | 0.52 | **yes** |
| MPPB (mouse plasma protein binding) | 1,756 | +0.23 | 2e-22 | +0.49 | 0.39 | **yes** |
| MGMB | 431 | +0.15 | 2e-03 | +0.15 | 0.23 | **yes** |
| LogD | 7,309 | −0.14 | 3e-31 | −0.20 | 0.12 | **yes** |
| HLM CLint (human microsomal clearance) | 4,541 | −0.10 | 1e-11 | +0.01 | 0.07 | **yes** |
| MBPB (mouse blood protein binding) | 1,426 | +0.03 | 0.32 | +0.23 | 0.24 | no |

The strongest drift is in the Caco-2 permeability assays, solubility, and mouse liver clearance. MBPB shows no reliable monotonic trend (p = 0.32). HLM clearance clears the bar statistically but is the weakest real effect (correlation −0.10, near-zero median change).

### 2. The time-split adds structural novelty, not value extrapolation

| Split / assay | Structural novelty (median 1-NN distance) | Value extrapolation (% test beyond training range) |
|---|---|---|
| **Time-split, per assay** | | |
| MPPB | 0.39 | 0.4% |
| Caco-2 Papp A>B | 0.38 | 0.1% |
| Caco-2 Efflux | 0.38 | 0.9% |
| MLM CLint | 0.36 | 0.0% |
| HLM CLint | 0.36 | 0.0% |
| KSOL | 0.35 | 0.4% |
| LogD | 0.35 | 0.0% |
| MBPB | 0.32 | 0.0% |
| MGMB | 0.27 | 0.0% |
| **Strategy anchors (LogD)** | | |
| Cluster-split | 0.41 | 0.0% |
| Time-split | 0.35 | 0.0% |
| Target-value split | 0.26 | **91.5%** |

The two axes are clearly distinct. The cluster-split maximizes structural novelty (1-NN ≈ 0.41) while keeping every test value inside the seen range (0% beyond). The target-value split is the mirror image: structurally it is the *closest* to training (1-NN ≈ 0.26, because similar values tend to come from similar molecules), yet 91.5% of its test values fall outside the training range — it is the only split that actually tests value extrapolation. The time-split sits between the two on structure and, like the cluster-split, induces essentially no value extrapolation (≤0.9% across all assays).

## Plots

<img width="2241" height="1627" alt="Image" src="https://github.com/user-attachments/assets/9971f160-c32e-4aa5-acfb-f7a0a81f41ce" />

 — per-assay value distribution across the five temporal windows (earliest→latest); drifting assays in coral, stable in gray

<img width="2223" height="878" alt="Image" src="https://github.com/user-attachments/assets/1a26a06f-3dc6-4509-b7f9-67ae725e320b" />

— structural novelty (x) vs value extrapolation (y); left: time-split per assay, right: the three strategies on LogD as anchors

## Reproduce

```bash
pixi run -e cheminformatics python notebooks/2.06-seal-split-quality.py
```

## Conclusion

Label drift over time is real and assay-specific (8 of 9 assays; MBPB is the exception), so a prospective time-split is a meaningful test of robustness to changing measurement distributions. But that is a different challenge from value extrapolation: in this dataset the time-split keeps test molecules inside the seen value range and only modestly increases structural novelty, whereas pushing models beyond the seen value range is exercised specifically by the target-value split. The two should be reported as separate generalization axes rather than conflated.

## Source

`notebooks/2.06-seal-split-quality.py`
